### PR TITLE
[MRG] Successive halving continued

### PR DESCRIPTION
**Pull request by NicolasHug**
- [X] Add a user guide section
- [X] Write docstrings (there are still things that need to be double checked though)
- [X] Heatmap example with comparison with GridSearchCV
- [X] iteration scores example

### CODE REVIEW

**Comment by amueller** [comment]
this diff looks weird. Btw, just changed the library/repo name.

**Comment by NicolasHug** [comment]
>this diff looks weird

Should be fixed

>Btw, just changed the library/repo name.

Ah, sad

**Comment by NicolasHug** [comment]
blah

This the output of the comparison with gridsearch, the other example is exactly the same except we don't run grid search. Should I merge the 2 examples and add a --run_grid_search option?

Code might be a bit more complex so I guess it depends wether we want our example to show how "easily" things can be done.

**Comment by amueller** [comment]
too many student suicides for fml to feel right...

**Comment by NicolasHug** [comment]
yeah makes sense

**Comment by amueller** [comment]
maybe combine them and always run gridsearch? would be nice to have the same color scale on both but otherwise looks cool.

**Comment by NicolasHug** [comment]
> maybe combine them and always run gridsearch?

Done 


sh

**Comment by amueller** [comment]
sorry I just reorganized everything lol

**Comment by amueller** [comment]
This probably needs a separate page in the user guide

File: doc/user_guide.rst [hunk 1]
```diff
@@ -119,6 +86,232 @@ performance.
...
+    .. [1] K. Jamieson, A. Talwalkar,
+       `Non-stochastic Best Arm Identification and Hyperparameter
+       Optimization <http://proceedings.mlr.press/v51/jamieson16.html>`_, in
+       proc. of Machine Learning Research, 2016.
+    .. [2] L. Li, K. Jamieson, G. DeSalvo, A. Rostamizadeh, .A Talwalkar,
+       `Hyperband: A Novel Bandit-Based Approach to Hyperparameter Optimization
+       <https://arxiv.org/abs/1603.06560>`_, in Machine Learning Research
+       18, 2018.
+
+Philosophy
```
Comment (amueller) [inline doc/user_guide.rst:280]: this should stay on top ;) and the halving stuff should go in a separate file

File: dabl/__init__.py [hunk 2]
```diff
@@ -2,7 +2,8 @@
 from .models import SimpleClassifier, SimpleRegressor
 from .plot.supervised import plot_supervised
 from .explain import explain
+from .search import GridSuccessiveHalving, RandomSuccessiveHalving
 
 __all__ = ['EasyPreprocessor', 'SimpleClassifier', 'SimpleRegressor',
-           'explain', 'clean',
-           'detect_types', 'plot_supervised']
+           'explain', 'clean', 'detect_types', 'plot_supervised',
+           'GridSuccessiveHalving', 'RandomSuccessiveHalving']
```
Comment (amueller) [inline dabl/__init__.py:9]: I wouldn't put these into the top level for now.

**Review by amueller — COMMENTED** [review]
lol I posted these earlier but looks like I didn't press "comment", sorry.

**Comment by NicolasHug** [comment]
Closing for #32 , comments were addressed

**Pull request closed without merging**